compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeBridge.desktop.kt

```diff
@@ -279,9 +279,43 @@ internal abstract class ComposeBridge(
         })
     }
 
+    // Decides which AWT events should be delivered, and which should be filtered out
+    private val awtEventFilter = object {
+
+        var isPrimaryButtonPressed = false
+
+        fun shouldSendMouseEvent(event: MouseEvent): Boolean {
+            // AWT can send events after the window is disposed
+            if (isDisposed)
+                return false
+
+            // Filter out mouse events that report a change in the pressed state of the primary
+            // button, but aren't themselves mouse press/release events.
+            // See https://github.com/JetBrains/compose-multiplatform/issues/2850 for details
+            val eventReportsPrimaryButtonPressed =
+                (event.modifiersEx and MouseEvent.BUTTON1_DOWN_MASK) != 0
+            if (eventReportsPrimaryButtonPressed != isPrimaryButtonPressed) {
+                isPrimaryButtonPressed = when (event.id) {
+                    MouseEvent.MOUSE_PRESSED -> true
+                    MouseEvent.MOUSE_RELEASED -> false
+                    else -> return false  // Note the return
```

Comment:
We didn't take into account one thing. On Windows, we can miss the released event:

```
1. open the application
2. press the button
3. press alt-tab to switch to another application
4. release the button
5. return to the application
6. application didn't receive the release event
```
If we merge this PR, the UI will be stuck after this sequence.

I checked it with this code:
```
import java.awt.Dimension
import java.awt.event.MouseAdapter
import java.awt.event.MouseEvent
import javax.swing.JFrame
import javax.swing.SwingUtilities

fun main() = SwingUtilities.invokeLater {
    val window = JFrame()
    window.size = Dimension(200, 200)
    window.addMouseListener(object : MouseAdapter() {
        override fun mousePressed(e: MouseEvent?) {
            println("mousePressed")
        }

        override fun mouseReleased(e: MouseEvent?) {
            println("mouseReleased")
        }
    })
    window.isVisible = true
}
```

Reply:
But actually, I'm not sure what we would want to happen in this case on Windows. If we don't filter the event, as soon as you move the mouse after returning to the window, the button will detect a "tap". If we do filter the event, the app will be "stuck" but only until the next time you press the button.

To fix this we actually need to fix `AwaitPointerEventScope.waitForUpOrCancellation`. Currently it does:
```
        val event = awaitPointerEvent(pass)
        if (event.changes.fastAll { it.changedToUp() }) {
            // All pointers are up
            return event.changes[0]
        }

        if (event.changes.fastAny {
                it.isConsumed || it.isOutOfBounds(size, extendedTouchPadding)
            }
        ) {
            return null // Canceled
        }
```
and the fix would be to check `isOutOfBounds` in the first condition too.